src/m365/spo/commands/file/file-retentionlabel-remove.ts

```diff
@@ -0,0 +1,158 @@
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.fileId &&
+          !validation.isValidGuid(args.options.fileId as string)) {
+          return `${args.options.fileId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['fileUrl', 'fileId'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (args.options.confirm) {
+      await this.removeFileRetentionLabel(logger, args);
+    }
+    else {
+      const result = await Cli.prompt<{ continue: boolean }>({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove the retentionlabel from file ${args.options.fileId || args.options.fileUrl} located in site ${args.options.webUrl}?`
+      });
+
+      if (result.continue) {
+        await this.removeFileRetentionLabel(logger, args);
+      }
+    }
+  }
+
+  private async removeFileRetentionLabel(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Removing retention label from file ${args.options.fileId || args.options.fileUrl} in site at ${args.options.webUrl}...`);
+    }
+    try {
+      const fileProperties = await this.getFileProperties(args);
+      const options: SpoListItemRetentionLabelRemoveCommandOptions = {
+        webUrl: args.options.webUrl,
+        listUrl: fileProperties.listServerRelativeUrl,
+        listItemId: fileProperties.id,
+        confirm: true,
+        output: 'json',
+        debug: this.debug,
+        verbose: this.verbose
+      };
+
+      await Cli.executeCommandWithOutput(SpoListItemRetentionLabelRemoveCommand as Command, { options: { ...options, _: [] } });
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getFileProperties(args: CommandArgs): Promise<{ id: string, listServerRelativeUrl: string }> {
+    const requestOptions: AxiosRequestConfig = {
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    if (args.options.fileId) {
+      requestOptions.url = `${args.options.webUrl}/_api/web/GetFileById('${args.options.fileId}')?$expand=ListItemAllFields`;
+    }
+    else {
+      requestOptions.url = `${args.options.webUrl}/_api/web/GetFileByServerRelativeUrl('${formatting.encodeQueryParameter(args.options.fileUrl!)}')?$expand=ListItemAllFields`;
+    }
+
+    const response = await request.get<FileProperties>(requestOptions);
+    return { id: response.ListItemAllFields.Id, listServerRelativeUrl: this.getListServerRelativeUrl(response.ServerRelativeUrl) };
+  }
+
+  private getListServerRelativeUrl(fileUrl: string): string {
+    return fileUrl.replace(/\/[^\/]+$/, '');
```

Comment:
This does not get you the `listServerRelativeUrl` perse, what if I execute the command on a file in a folder?  

You should call the endpoint slightly different to include the right information:

`https://contoso.sharepoint.com/sites/sales/_api/Web/GetFileById(guid'96471462-ab76-4ef6-837a-2e9e1aba1052')?$expand=ListItemAllFields,ListItemAllFields/ParentList&$select=ListItemAllFields/ParentList/Id,ListItemAllFields/Id`

This will get you the list item id and list id.